sky/skylet/configs.py

```diff
@@ -25,20 +25,37 @@ def _safe_cursor():
         conn.close()
 
 
-with _safe_cursor() as c:  # Call it 'c' to avoid pylint complaining.
-    c.execute("""\
-        CREATE TABLE IF NOT EXISTS config (
-            key TEXT PRIMARY KEY,
-            value TEXT)""")
+def ensure_table(func: callable):
+    """Ensure the table exists before calling the function."""
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        global _table_created
+        if not _table_created:
+            with _safe_cursor(
+            ) as c:  # Call it 'c' to avoid pylint complaining.
+                # Use WAL mode to avoid locking problem in #1507.
+                # Reference: https://stackoverflow.com/a/39265148
+                c.execute('PRAGMA journal_mode=WAL')
+                c.execute("""\
+                    CREATE TABLE IF NOT EXISTS config (
+                        key TEXT PRIMARY KEY,
+                        value TEXT)""")
+        _table_created = True
```

Comment:
Will this have a race?

Reply:
This variable is only a weak guard to reduce the overhead for calling these commands every time the function in `configs.py` is called in the same SkyPilot process. The main effective part for avoid concurrent sqlite commands call is to avoid running the commands every time the `configs.py` is imported. 
Race condition: I don't think it matters much, because
1. This variable is a weak guard and does shared across different SkyPilot processes launched by different CLIs.
1. We will not switch it back to False, so even if there are multiple threads calling functions in `configs.py`, only the first few calls will run the sqlite commands together, which might be fine.